test/onnx/test_utility_funs.py

```diff
@@ -1044,6 +1044,137 @@ def forward(self, input, other):
         iter = graph.nodes()
         self.assertEqual(next(iter).kind(), "custom_namespace::custom_op")
 
+    @skipIfUnsupportedMinOpsetVersion(11)
+    def test_autograd_layernorm_shape(self):
+        def symbolic_pythonop(ctx: torch.onnx.SymbolicContext, g, *args, **kwargs):
+            return g.op("com.microsoft::PythonOp")
+
+        register_custom_op_symbolic("prim::PythonOp", symbolic_pythonop, 1)
+        self.addCleanup(unregister_custom_op_symbolic, "prim::PythonOp", 1)
+
+        # necessay parameters for transformer embeddings
+        hidden_size, vocab_size, max_position_embeddings, type_vocab_siz, batch_size = (
+            768,
+            128000,
+            512,
+            0,
+            16,
+        )
+
+        # wrap nn.layernorm into autograd.func repro the same
+        # issue as using apex fusedlayernorm
+        class CustomLayerNorm(torch.autograd.Function):
+            @staticmethod
+            def forward(ctx, embedding):
+                ctx.save_for_backward(embedding)
+                layer_norm = torch.nn.LayerNorm(hidden_size, eps=1e-12)
+                return layer_norm(embedding)
+
+        class EmbeddingModule(torch.nn.Module):
```

Comment:
Possible to reduce the size of the test (in terms of the test model) to make it clearer?

Reply:
+1 on this. I'd recommend finding the simplest repro code to capture the exact failing pattern.

Reply:
+1. Check on the torch ir which layer emits the transpose after the python and remove the other layers that come before/after and see if that repros the bug. There must be a layer consuming the transpose, IIRC

Reply:
I did try with only layer_norm and transpose but it didn't work, as onnx shape inference did not take it as static for some reasons that I don't know. I guess I will try hijack the result from BertEmbedding and use it directly as input for layer_norm

Reply:
Having some changes on the repro:
1. move it from `test_utility_func` to `test_pytorch_onnx_shape_inference`
2. deleted unrelated modules (torch.nn.Embeddings and torch.nn.linear)
3. Simplify the test also with using `Reshape` to consume the static batch_size
4. assign magic number 3 with a meaningful name `dim`